src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,189 @@
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo  myErrors(JsonMappingException exception) {
+	    logger.info("----Caught IOException----\n"+request.getDescription(false)+"\n Detail IOException:"+exception.getMessage()+exception.getStackTrace());
+	    return new ErrorInfo(request.getContextPath(), "General Exception",HttpStatus.CONFLICT.toString());
+      
+	  }
+
+ 	 
+	  @ResponseStatus(HttpStatus.CONFLICT)
+	  @ExceptionHandler(Exception.class)
+	  @ResponseBody
+	  /**
+	   * Handles General Exception
+	   * @param exception
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo  myError(Exception exception) {
+	    logger.info("----Caught Exception----\n"+request.getDescription(false)+"\n Detail IOException:"+exception.getMessage()+exception.getStackTrace());
+	    return new ErrorInfo(request.getContextPath(), "General Exception",HttpStatus.CONFLICT.toString());
+        
+	  }
+	  
+	  @ResponseStatus(HttpStatus.BAD_REQUEST)
+	  @ExceptionHandler(IllegalArgumentException.class)
+	  @ResponseBody
+	  /**
+	   * Handles IllegalArgumentException
+	   * @param IllegalArgumentException
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo illegal(IllegalArgumentException exception) {
+            logger.info("----This is a ilegal argument exception ----\n"+request.getDescription(false)+"\n Detail IllegalArgumentException:"+exception.getMessage()+exception.getStackTrace());
+            return new ErrorInfo(request.getContextPath(), "IllegalArgumentException",HttpStatus.BAD_REQUEST.toString());
+            
+	  }
+	  
+	  @ResponseStatus(HttpStatus.NOT_FOUND)
+	  @ExceptionHandler(KeyWordNotFoundException.class)
+	  @ResponseBody
+	  /***
+	   * Handles KeywordNotFoundException
+	   * @param exception
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo notFound(Exception exception) {
+            logger.info("----Caught KeywordNotFoundException----\n"+request.getDescription(false)+"\n Detail NotFoundException:"+exception.getMessage()+exception.getStackTrace());
+            return new ErrorInfo(request.getContextPath(), "KeyWordNotFoundException",HttpStatus.NOT_FOUND.toString());
+            
+	  }
+	  
+	  
+	  @ResponseStatus(HttpStatus.NOT_FOUND)
+	  @ExceptionHandler(ResourceNotFoundException.class)
+	  @ResponseBody
+	  /***
+	   * Handles ResourceNotFoundException
+	   * @param ResourceNotFoundException
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo resourceNotFound(ResourceNotFoundException exception) {
+            logger.info("----Caught KeywordNotFoundException----\n"+request.getDescription(false)+"\n Detail ResourceNotFoundException:"+exception.getMessage()+exception.getStackTrace());
+            return new ErrorInfo(request.getContextPath(), "Requested record/data not found.",HttpStatus.NOT_FOUND.toString());
+            
+	  }
+	  
+	  @ResponseStatus(HttpStatus.BAD_REQUEST)
```

Comment:
BAD_REQUEST (400) is for an error that is the client's fault--they called the service incorrectly.  An IOException sounds like a server problem (e.g. the storage system had a read error), yes?  This should result in `HttpStatus.INTERNAL_SERVER_ERROR`, yes?

Reply:
yes.

Reply:
Its just by mistake.